### PR TITLE
[skip ci] mod cowboy

### PR DESCRIPTION
**Pull request by goddammit**
please see if makes sense

This PR addresses #

Proposed changes include:
* describe the functionality changes
* describe new or updated tests
* describe changes to the documentation

### CODE REVIEW

File: doc/advanced-configuration/Listener-modules.md [hunk 1]
```diff
@@ -99,6 +99,10 @@ Unlike `ejabberd_c2s`, it doesn't use `ejabberd_receiver` or `ejabberd_listener`
 ### HTTP module: `mod_cowboy`
 
 This module provides an additional routing layer on top of HTTP(s) or WS(S) protocols.
+Regular cowboy handlers cannot cleanly handle both protocols on the same port.
```
Comment (studzien) [inline doc/advanced-configuration/Listener-modules.md]: I think it's perfectly fine to have a WS and HTTP handlers attached to the same listener (port) but different resources.
@michalwski,  wasn't it about the **same resource**?

Comment (michalwski) [reply doc/advanced-configuration/Listener-modules.md]: It is all possible only not in a clean way, which is sth subjective though. IMO clean cowboy handlers operates on single transport WS or HTTP (not both) and possible single or at least (similar resources). From what I understand `mod_cowboy` itself may not be clean but makes it easier and cleaner for the layers on top of it.

Comment (Nyco) [inline doc/advanced-configuration/Listener-modules.md]: It is "Cowboy" with an uppercase "C".

Comment (fenek) [reply doc/advanced-configuration/Listener-modules.md]: I think we should rephrase it into something like this:

It allows other HTTP/WS modules to coexist under the same URL on the single port.
Packets are forwarded to them based on protocol.
This mechanism is transparent for actual handlers so the path sharing does not require additional code.
Example: If you wish, you can use BOSH and WS XMPP handlers (`mod_bosh`, `mod_websockets`) on single port and URL without any code modifications.

What do you think?

Comment (michalwski) [reply doc/advanced-configuration/Listener-modules.md]: Yes, @fenek described it in the most accurate way in my opinion.

Comment (michalwski) [reply doc/advanced-configuration/Listener-modules.md]: IMO, it doesn't have to start with the uppercase "C". It's worth surrounding it with `` though like `cowboy`.

Comment (goddammit) [reply doc/advanced-configuration/Listener-modules.md]: It allows other HTTP/WS modules to coexist under the same URL on the single port.
Packets are forwarded to them based on the protocol.
This mechanism is transparent to actual handlers so the path sharing does not require any additional code.
Example: If you wish, you can use BOSH and WS XMPP handlers (mod_bosh, mod_websockets) on a single port and a URL without any code modifications.

**Pull request merged**